projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/properties/PropertiesServiceTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.properties;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.vmware.taurus.properties.service.JobProperties;
+import com.vmware.taurus.properties.service.PropertiesRepository;
+import com.vmware.taurus.properties.service.PropertiesService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class PropertiesServiceTest {
+
+  @Mock private PropertiesRepository propertiesRepository;
```

Comment:
but you are never checking that the repository works? 
Why mock it? if you don't mock it you will need to write less code and also test more.

Reply:
I tend to agree. 

I prefer (including in unit tests) to use test doubles/test substitute (fully functioning versions of the dependency) when possible and feasible than mock version. They make the test slightly more black box (hence less fragile, less false positives), make them easier to write (since you need not write a ton of logic to mock methods), and are better for complex interactions (that would be hard to mock).

Reply:
Hello,
I added these tests in the context of vep-1493-vault-integration - as a next step I intend to do some refactoring and extract a common interface for the Properties and the (upcoming) Secrets Service. 

I believe the test you are asking for is already present - DataJobPropertiesIT - however it's an integration test which runs slower and requires more infra setup.

Do you agree?